### PR TITLE
Add column descriptions to sample optimized tables

### PR DESCRIPTION
**Pull request by samanvp**
In this PR we copy column descriptions from variant optimized tables to sample optimized schema json file.

### CODE REVIEW

**Comment by samanvp** [comment]
This PR is ready to review now. Test runs show correct column descriptions for sample optimized tables.

File: gcp_variant_transforms/libs/partitioning.py [hunk 1]
```diff
@@ -270,6 +314,64 @@ def copy_to_flatten_table(self, output_base_table_id):
...
+    # type: (dict, dict, str) -> None
+    """Copies column descriptions from variant table to sample table schema.
+
+    Because of the way we create sample optimized tables schema, we miss the
+    column descriptions:
+      * Compose a query to flatten call column.
+      * Write the output of the query (LIMIT 1) to a temporary table.
+      * Extract the temporary table's schema into a json file.
+
+    This class enables use to add 'description' fields to the json schema file.
```
Comment (tneymanov) [inline gcp_variant_transforms/libs/partitioning.py:330]: s/use/us

File: gcp_variant_transforms/libs/partitioning.py [hunk 2]
```diff
@@ -270,6 +314,64 @@ def copy_to_flatten_table(self, output_base_table_id):
...
 
 
+class ParseTableSchema(object):
+  """Parses json file containing BigQuery table schema adds description to it"""
+
+  def __init__(self, schema_dict, description_dict, flattened_col=''):
+    # type: (dict, dict, str) -> None
+    """Copies column descriptions from variant table to sample table schema.
+
+    Because of the way we create sample optimized tables schema, we miss the
```
Comment (tneymanov) [inline gcp_variant_transforms/libs/partitioning.py:324]: s/tables/table, or remove schema

File: gcp_variant_transforms/libs/partitioning.py [hunk 3]
```diff
@@ -270,6 +314,64 @@ def copy_to_flatten_table(self, output_base_table_id):
...
+
+  def _get_name(self, item):
+    field_name = item.get('name')
+    if self._is_schema_flatten:
+      field_name = field_name.replace(self._flattened_col_prefix,
+                                      self._flattened_col_path, 1)
+    return field_name
+
+  def _parse_schema(self, path, schema_dict):
+    for item in schema_dict:
```
Comment (tneymanov) [inline gcp_variant_transforms/libs/partitioning.py:360]: Will this change the order of the fields in the scheme? If so, can it affect the vcf-to-bq-to-vcf pipeline (ie, original input not matching output)?

File: gcp_variant_transforms/libs/partitioning.py [hunk 4]
```diff
@@ -113,11 +118,27 @@ def _run_query(self, query):
...
+      for row in result:
+        result_list.append(str(row.values()[0]))
+      return result_list
+    elif len(result.schema) == 2:
+      result_dict = {}
+      for row in result:
+        result_dict[row.values()[0]] = row.values()[1]
+      return result_dict
+    else:
+      ValueError('This function can only handle results with 1 or 2 columns.')
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/libs/partitioning.py]: I updated the error message to make this more clear.

File: gcp_variant_transforms/libs/partitioning.py (hunk 2 above)
Comment (samanvp) [reply gcp_variant_transforms/libs/partitioning.py]: done.

File: gcp_variant_transforms/libs/partitioning.py (hunk 1 above)
Comment (samanvp) [reply gcp_variant_transforms/libs/partitioning.py]: done.

File: gcp_variant_transforms/libs/partitioning.py (hunk 3 above)
Comment (samanvp) [reply gcp_variant_transforms/libs/partitioning.py:361]: No, we don't modify the oder or items in the list, we only add to each item `item['description']` field.

**Pull request closed without merging**